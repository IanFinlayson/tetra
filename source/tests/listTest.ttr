def main():
  bob = 5
  print("set bob = 5 and bob = ")
  print(bob)
  print("\n")
  l = [bob]
  print("set l = [bob] and l = ")
  print(l)
  print("\n")
  l[0] = 1
  print("set l[0] = 1 and l = ")
  print(l)
  print("\n")
  print("now bob = ")
  print(bob)
  print("\n")

  m = [6]
  print("set m = [6] and m = ")
  print(m)
  print("\n")
  n = [7]
  print("set n = [7] and n = ")
  print(n)
  print("\n")
  o = [m,n]
  print("set o = [m,n] and o = ")
  print(o)
  print("\n")
  m[0] = 2
  print("set m[0] = 2 and m = ")
  print(m)
  print("\n")
  print("now o = ")
  print(o)
  print("\n")

  p = l
  print("set p = l and p = ")
  print(p)
  print("\n")
  p[0] = 3  
  print("set p[0] = 3 and p = ")
  print(p)
  print("\n")
  print("now l = ")
  print(l)
