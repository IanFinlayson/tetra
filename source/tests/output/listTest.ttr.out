set bob = 5 and bob = 5
set l = [bob] and l = [5]
set l[0] = 1 and l = [1]
now bob = 5
set m = [6] and m = [6]
set n = [7] and n = [7]
set o = [m,n] and o = [[6], [7]]
set m[0] = 2 and m = [2]
now o = [[2], [7]]
set p = l and p = [1]
set p[0] = 3 and p = [3]
now l = [3]
